src/main/java/no/idporten/userservice/data/UserService.java

```diff
@@ -4,59 +4,86 @@
 import org.springframework.stereotype.Service;
 import org.springframework.util.Assert;
 
-import java.util.*;
+import java.time.Instant;
+import java.util.Collections;
+import java.util.List;
+import java.util.Optional;
+import java.util.UUID;
 
 @RequiredArgsConstructor
 @Service
 public class UserService {
 
     private final UserRepository userRepository;
 
-    public IDPortenUser findUser(String id) {
-        Optional<UserEntity> user = userRepository.findByUuid(UUID.fromString(id));
-        if(user.isEmpty()){
+    public IDPortenUser findUser(UUID uuid) {
+        Optional<UserEntity> user = userRepository.findByUuid(uuid);
+        if (user.isEmpty()) {
             return null;
         }
-        return convert(user.get());
+        return new IDPortenUser(user.get());
     }
 
-    public List<IDPortenUser> searchForUser(String pid) {
-        Optional<UserEntity> users = userRepository.findByPersonIdentifier(pid);
-        return users.stream().map(this::convert).toList();
+    public List<IDPortenUser> searchForUser(String personIdentifier) {
+        Optional<UserEntity> users = userRepository.findByPersonIdentifier(personIdentifier);
+        if (users.isEmpty()) {
+            return Collections.EMPTY_LIST;
+        }
+        return users.stream().map(IDPortenUser::new).toList();
     }
 
     public IDPortenUser createUser(IDPortenUser idPortenUser) {
         Assert.isNull(idPortenUser.getId(), "id is assigned by server");
         Assert.isTrue(searchForUser(idPortenUser.getPid()).isEmpty(), "User exists");
 
-        UserEntity userSaved = userRepository.save(copyData(idPortenUser));
-        return convert(userSaved);
+        UserEntity userSaved = userRepository.save(idPortenUser.toEntity());
+        return new IDPortenUser(userSaved);
     }
 
-    public IDPortenUser updateUser(String id, IDPortenUser idPortenUser) {
-        Assert.notNull(id, "id is mandatory");
-        Assert.isTrue(Objects.equals(id, idPortenUser.getId().toString()), "id must match resource.id");
-        UserEntity savedUser = userRepository.save(copyData(idPortenUser));
-        return convert(savedUser);
+    public IDPortenUser updateUser(IDPortenUser idPortenUser) {
+        Assert.notNull(idPortenUser.getId(), "id is mandatory");
+        UserEntity userToSave = idPortenUser.toEntity();
+        UserEntity savedUser = userRepository.save(userToSave);
```

Comment:
toEntity() vil ikke fungere her.  Her tror jeg vi må bestemme oss for hvilke attributter det er tillatt å oppdatere, og så kopiere disse over på entity som vi søkte opp.